polaris/hub/client.py

```diff
@@ -497,6 +504,7 @@ def upload_dataset(
         access: AccessType = "private",
         timeout: TimeoutTypes = (10, 200),
         owner: Optional[Union[HubOwner, str]] = None,
+        on_conflict: ZarrConflictResolution = "raise",
```

Comment:
To minimize friction for people that are familiar with Zarr, I suggest we keep the name the same as for Zarr: `if_exists`.